web/access_control.go

```diff
@@ -132,6 +132,25 @@ func (s *AutograderService) isTeacher(userID, courseID uint64) bool {
 	})
 }
 
+// changingTeacherState returns true if the method will attempt change on a teaher enrollment
+func (s *AutograderService) changingTeacherState(enrol *pb.Enrollment) bool {
+	currEnrol, _ := s.db.GetEnrollmentByCourseAndUser(enrol.GetCourseID(), enrol.GetUserID())
+
+	if (currEnrol.GetStatus() == pb.Enrollment_TEACHER) && (enrol.GetStatus() != pb.Enrollment_TEACHER) {
+		return true
+	}
+	return false
+}
+
+// isCourseCreator returns true if the given user is course creator for the given course
+func (s *AutograderService) isCourseCreator(courseID, userID uint64) bool {
+	course, _ := s.db.GetCourse(courseID, false)
```

Comment:
GetCourse can return `nil`; so here you should also check for `err != nil` and if so, just return false.

Reply:
Or alternatively, check that `course != nil` before using it for `GetCourseCreatorID()`.

Reply:
But if `course` is nil, `course.GetCourseCreatorID()` should return 0, shouldn't it?

Reply:
That is correct. So this should work fine.